### DISCUSSION TITLE
Implementing Example web3wallet: "session_proposal" not triggering.

### DISCUSSION THREAD

**Discussion by ph101pp** [WalletConnect/walletconnect-monorepo#3711] [Developer Support]
Hi,

I'm trying to implement a simple web3wallet example that gets injected via chrome extension. 
The goal is to add this to our existing wallet extension and get it added to the explorer and web3Modal so users can select our wallet from the list.

I'm mainly following these examples:
- https://docs.walletconnect.com/2.0/web3wallet/wallet-usage
- https://github.com/WalletConnect/web-examples/blob/371e4f334d6ef042bade58a1fd67629adf3bda86/wallets/react-wallet-v2/src/utils/WalletConnectUtil.ts#L6C2-L18C5
- https://github.com/WalletConnect/web-examples/blob/371e4f334d6ef042bade58a1fd67629adf3bda86/wallets/react-wallet-v2/src/hooks/useWalletConnectEventsManager.ts#L18

So far I've got a projectId and this code that is injected:

```javascript
async function initWalletConnect() {
    const core = new Core({ projectId })
    const web3wallet = await Web3Wallet.init({
      core,
      metadata: {
        name: 'React Wallet Example',
        description: 'React Wallet for WalletConnect',
        url: 'https://walletconnect.com/',
        icons: ['https://avatars.githubusercontent.com/u/37784886']
      }
    });

    web3wallet.on("session_proposal", async sessionProposal => {
      console.log("pca", "session_proposal", sessionProposal);
      // connect to wallet and approve session if successful.
    })

    web3wallet.on("session_request", async sessionRequest=> {
      console.log("pca", "session_request", sessionRequest)
      // pass request to wallet and return result
    })

    // @ts-ignored
    window.pcaConnect = (uri) => {
      return web3wallet.pair({ uri });
    }
}
```
These are the package versions I'm using:
```
    "@walletconnect/core": "^2.10.1",
    "@walletconnect/web3wallet": "^1.9.1",
```

I'm getting a pair uri (`ws://valid_uri`) from here: `https://react-app.walletconnect.com/`

My assumption would be that calling: `web3wallet.pair({ uri: "ws://valid_uri" })` should trigger the `session_proposal` event handler. 
The script gets injected and executed and calling my test method from the console: `window.pcaConnect("ws://valid_uri")` results in a fulfilled promise but no events.

What am I missing? Does my projectId need to be approved?

**Comment by pitasi** [comment]
Same here! I wanted to build a slim react app just as a proof of concept, similar to the example's one.

I also see this in the console:

```
GET https://verify.walletconnect.com/attestation/6bffe... 404 (Not Found)
```

Keep in mind that I'm testing from localhost (but I read in another answer that it should be working).

**Reply by pitasi** [reply]
related code: https://github.com/WalletConnect/walletconnect-monorepo/blob/d59d0f73483117b4faea72f1921fcc59a9407d78/packages/sign-client/src/controllers/engine.ts#L748-L749

**Comment by indytheoracle** [comment]
I'm having the same issue. It seems like a very basic example as the one [ph101pp](https://github.com/ph101pp) posted should work, but no events get fired for me either.

**Reply by indytheoracle** [reply]
I created a starter repository with a vite/react/walletconnect template and nothing else. Please see it here. Its easy to setup and run quickly. https://github.com/indytheoracle/vite-template-react-walletconnect

**Reply by pitasi** [reply]
it must have something to do with Vite in dev mode and React that calls hooks twice

**Comment by indytheoracle** [comment]
I got it working here: https://github.com/indytheoracle/vite-template-react-walletconnect
